### DISCUSSION TITLE
Which resource should I use for pull replication?

### DISCUSSION THREAD

**Discussion by anstrom** [jfrog/terraform-provider-artifactory#841] [General] [answered]
We want to configure pull replication using Terraform, and I'm confused after reading the documentation.

There seems to be overlapping config settings between the resources `remote_repository_replication` and `pull_replication`.

https://registry.terraform.io/providers/jfrog/artifactory/latest/docs/resources/remote_repository_replication
https://registry.terraform.io/providers/jfrog/artifactory/latest/docs/resources/pull_replication

The two resources have included official documentation for pull replication in the description, but `remote_repository_replication` has documentation for push replication in the __Imports__ section.

**Comment by alexhung** [comment]
@anstrom Thanks for the report. The documentation for the replication resources are not super clear, I admit. I'll add this to our plan to improve them.

**Reply by alexhung** [reply, marked as answer]
To answer your question though, `remote_repository_replication` resource is the one to use. The other ones are old and deprecated. See [change log of v7.3.0](https://github.com/jfrog/terraform-provider-artifactory/blob/master/CHANGELOG.md#730-march-17-2023-tested-on-artifactory-7558)

Deprecation messages were added to the resources (i.e. the message will appear if you use them in your HCL and run `terraform plan`) but we forgot to include them in the documentation. I'll add them as part of the documentation improvement.

**Reply by anstrom** [reply]
Thanks for the clarification!
